### DISCUSSION TITLE
How to find how many indentation lines exists in a line?

### DISCUSSION THREAD

**Discussion by diogo-rossi** [microsoft/vscode-discussions#461] [Extension Development QnA] [answered]
In my extension, I'd like to find how many indentation lines exists in a generic line.

Is there a way to get that?

For instance:
Given the following python code

* Lines `2`,`3`,`4`,`5` have only 1 indentation line
* Line `6` have 2 indentation lines
* Lines `7`,`8` have 3 indentation lines
* Lines `1`,`9`,`10`,`11` have no indentation line at all

![image](https://user-images.githubusercontent.com/57204023/218785862-9a1b1cb9-514a-4367-8635-1e07c43067af.png)

Is there a way to get that?

**Comment by isidorn** [comment, marked as answer]
Yes, in your extension you can get the content of the editor, and then inspect the content and figure out indentation.

**Reply by diogo-rossi** [reply]
But the content of the line does not tell how many indentation **lines** exist in the line.
It could tell only whether the line has an indentation it self.

In the example above: line `7` is empty, but it has 3 indentation lines.

**Reply by isidorn** [reply]
Aha I understand. You could still look at the neighbouring lines and compute this somehow.
But, no there is no direct API for you to get this information.

**Reply by diogo-rossi** [reply]
Thanks for the answer.

Maybe there is a workaround. 

I could look at the neighbouring lines but it can become tricky.

Imagine if I put several empty lines between `if True:` (line `6`) and the `print('hello')` statement: all of then would have 3 indentation lines, but I would need to look for a "far neighbour line" to get the information.
